server/base/src/main/java/org/apache/accumulo/server/security/SecurityOperation.java

```diff
@@ -270,7 +270,7 @@ public Authorizations getUserAuthorizations(TCredentials credentials) throws Thr
   }
 
   public boolean userHasAuthorizations(TCredentials credentials, List<ByteBuffer> list) throws ThriftSecurityException {
-    authenticate(credentials);
+    // Authentication check not done here because this method is always called in conjunction with canScan that does auth check.
```

Comment:
I think this is a bit risky... somebody could add some code internally which uses this method without an authentication check (they might overlook the comment). Maybe a method rename, like `alreadyAuthenticatedUserHasAuthorizations` might make it more clear. It's a bit wordy, but my concern is that this method will be misused. Maybe there's another, simpler refactoring which would also avoid risky future misuse?

Reply:
I renamed the method and added javadoc.